http://bugs.gentoo.org/134547
http://gcc.gnu.org/PR29114

2006-11-19  John David Anglin  <dave.anglin@nrc-cnrc.gc.ca>

	PR target/29114
	* pa.c (emit_move_sequence): Don't split constants with PLUS for modes
	larger than BITS_PER_WORD.

--- branches/gcc-4_1-branch/gcc/config/pa/pa.c	2006/11/19 18:27:03	118994
+++ branches/gcc-4_1-branch/gcc/config/pa/pa.c	2006/11/19 19:06:09	118995
@@ -1879,6 +1879,7 @@
 	     because PLUS uses an 11-bit immediate and the insn sequence
 	     generated is not as efficient as the one using HIGH/LO_SUM.  */
 	  if (GET_CODE (operand1) == CONST_INT
+	      && GET_MODE_BITSIZE (mode) <= BITS_PER_WORD
 	      && GET_MODE_BITSIZE (mode) <= HOST_BITS_PER_WIDE_INT
 	      && !insert)
 	    {
